LiteDB/Engine/Disk/DiskWriterQueue.cs

```diff
@@ -88,19 +88,17 @@ private async Task ExecuteQueue()
                     }
                     else
                     {
-                        lock (_queueSync)
-                        {
-                            if (_queue.Count > 0) continue;
 
-                            _queueIsEmpty.Set();
-                            _queueHasItems.Reset();
+                        if (_queue.Count > 0) continue;
+
+                        _queueIsEmpty.Set();
+                        _queueHasItems.Reset();
 
-                            if (_shouldClose) return;
-                        }
+                        if (_shouldClose) return;
 
                         _stream.FlushToDisk();
 
-                        await _queueHasItems.WaitAsync();
+                        _queueHasItems.WaitAsync().GetAwaiter().GetResult();
```

Comment:
Why not a normal ManualResetEvent then?

Reply:
Probably, but I don't want to change it too much. Substantial changes will happen on #2505